### PR TITLE
chore(blobby): penalize non-batched traffic for overflow detection

### PR DESCRIPTION
**Pull request by xvello**
## Problem

Sessions from very old posthog-js versions can still slow down a partition without triggering overflow, because they emit one message per event, instead of batching them together.

Batching was implemented in https://github.com/PostHog/posthog-js/pull/694, released as v1.68.2 in 2023. Default batching settings are a linger of 2 seconds, or a size of 900kib

## Changes

- `OverflowManager` has a new `minPerCall` property. Calls to `observe` consume at least this quantity
- The value is set to 1MB/session/batch on blobby (after messages in the same Kakfa batch are aggregated). This will make a session trigger overflow if it sends more than 5 batches per functional second after a 200 burst budget. Might not be too high? We'll see and increase if needed.

### CODE REVIEW

**Comment by bretthoerner** [comment]
```
FAIL tests/main/ingestion-queues/session-recording/session-recordings-consumer.test.ts (8.895 s)
  ● ingester with consumeOverflow=false › when ingester.stop is called in teardown › overflow detection › should not trigger overflow if under threshold
```

Seems likely related?

File: plugin-server/tests/main/ingestion-queues/session-recording/session-recordings-consumer.test.ts [hunk 1]
```diff
@@ -597,7 +598,9 @@ describe.each([[true], [false]])('ingester with consumeOverflow=%p', (consumeOve
         describe(
             'overflow detection',
             consumeOverflow
-                ? () => {} // Skip these tests when running with consumeOverflow (it's disabled)
```
Comment (xvello) [inline plugin-server/tests/main/ingestion-queues/session-recording/session-recordings-consumer.test.ts:600]: new eslint rule

**Review by bretthoerner — APPROVED** [review]
Good idea!

I don't have a good fell for whether 1MB each is low or high, it sounds like you think it may not be high enough? If so that seems like a safer bet to start. 👍

**Review by pauldambra — APPROVED** [review]
👍

**Comment by xvello** [comment]
Should be merged indeed, was not sure about the value to set though.

**Pull request merged**